netket/_scipy_integrators.py

```diff
@@ -0,0 +1,131 @@
+import scipy
+import numpy as np
+from scipy.integrate import OdeSolver, DenseOutput
+from warnings import warn
+
+
+def validate_first_step(first_step, t0, t_bound):
+    """Assert that first_step is valid and return it."""
+    if first_step <= 0:
+        raise ValueError("`first_step` must be positive.")
+    if first_step > np.abs(t_bound - t0):
+        raise ValueError("`first_step` exceeds bounds.")
+    return first_step
+
+
+def warn_extraneous(extraneous):
```

Comment:
```suggestion
def _warn_extraneous(extraneous):
```
(And corresponding changes below.)